### PR TITLE
Refactor lists

### PR DESCRIPTION
**Pull request by sologeek**

### CODE REVIEW

File: ribir/examples/storybook.rs [hunk 1]
```diff
@@ -87,13 +89,52 @@ fn main() {
...
+                ListItem {
+                  Leading { svgs::MENU }
+                  HeadlineText(Label::new("One line list item"))
+                  Trailing { Label::new("100+") }
+                }
+                Divider { indent: 16. }
+                ListItem {
+                  line_number: 2,
+                  Leading {
+                    IntoWidget::into_widget(
```
Comment (M-Adoo) [inline ribir/examples/storybook.rs:117]: Avatar ?

File: widgets/src/avatar.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  fn compose_child(_: State<Self>, child: Self::Child) -> Widget {
+    widget! {
+      init ctx => {
+        let AvatarStyle {
+          size, radius, background, text_style, text_color,
+        } = AvatarStyle::of(ctx).clone();
+      }
+      SizedBox {
+        size,
+        DynWidget {
```
Comment (M-Adoo) [inline widgets/src/avatar.rs:44]: widget::from better ?

File: widgets/src/lists.rs [hunk 3]
```diff
@@ -1,126 +1,409 @@
...
+///     }
+///     // use leading label
+///     ListItem {
+///       Leading { Label::new("A") }
+///       HeadlineText(Label::new("headline text"))
+///     }
+///     // use leading custom widget
+///     ListItem {
+///       Leading {
+///         IntoWidget::into_widget(
```
Comment (M-Adoo) [inline widgets/src/lists.rs:62]: unnecessary IntoWidget wrap.

File: widgets/src/lists.rs [hunk 4]
```diff
@@ -1,126 +1,409 @@
...
+///     // use trailing label
+///     ListItem {
+///       HeadlineText(Label::new("headline text"))
+///       Trailing { Label::new("A") }
+///     }
+///     // use trailing custom widget
+///     ListItem {
+///       HeadlineText(Label::new("headline text"))
+///       Trailing {
+///         IntoWidget::into_widget(
```
Comment (M-Adoo) [inline widgets/src/lists.rs:97]: Unnecessary `IntoWidget` wrap.

File: widgets/src/lists.rs [hunk 5]
```diff
@@ -1,126 +1,401 @@
...
+  Custom(DecorateTml<P, Widget>),
 }
 
-#[derive(Default, Declare)]
-pub struct ListItem;
+impl<P> EdgeWidget<P>
+where
+  P: TmlFlag + Default,
+{
+  pub fn into_widget(self, config: ListItemConfig) -> Widget {
```
Comment (M-Adoo) [inline widgets/src/lists.rs:208]: Unnecessary `pub`,

**Review by M-Adoo — APPROVED** [review]

**Review by wjian23 — APPROVED** [review]

**Pull request merged**